sonolark/lib/kube/kube.go

```diff
@@ -731,3 +741,79 @@ func (m *kubePackage) kubeGet(ctx context.Context, r *apiResource, wait time.Dur
 
 	// not reachable
 }
+
+func (m *kubePackage) kubePortForwardFn(t *starlark.Thread, b *starlark.Builtin, args starlark.Tuple, kwargs []starlark.Tuple) (starlark.Value, error) {
+	if len(args) != 0 {
+		return nil, fmt.Errorf("<%v>: positional args not supported by `kube.portforward': %v", b.Name(), args)
+	}
+
+	if len(kwargs) < 3 {
+		return nil, fmt.Errorf("<%v>: expected at least <resource>=<name> <localport>=<port> <podport>=<port>", b.Name())
+	}
+
+	resource, name, err := getResourceAndName(kwargs[0])
+	if err != nil {
+		return nil, fmt.Errorf("<%v>: %s", b.Name(), err.Error())
+	}
+
+	// If resource is not namespace itself (special case), attempt to parse
+	// namespace out of the arg value.
+	var namespace string
+	if resource != namespaceResrc {
+		ss := strings.Split(name, "/")
+		if len(ss) > 1 {
+			namespace = ss[0]
+			name = ss[1]
+		}
+	}
+
+	var localPort int64
+	var podPort int64
+	for _, kv := range kwargs[1:] {
```

Comment:
The unpacking of args can be simplified I think; see https://github.com/vmware-tanzu/sonobuoy-plugins/blob/3950b18526ee4da49f33385fb6f87acb25064339/sonolark/lib/kube/kube.go#L212-L219

I'm still new to lots of the starlark util funcs but I think this does the parsing that you want and makes it a little easier to read/modify